engine/runner/src/main/scala/org/enso/runner/Main.scala

```diff
@@ -760,7 +767,19 @@ object Main {
         case Some(main) if mainMethodName != "main" =>
           main.execute(mainType.newInstance())
         case Some(main) =>
-          val res = main.execute()
+          // Opportunistically parse arguments and convert to ints.
+          // This avoids conversions in main function.
+          val parsedArgs: Seq[AnyRef] = ArraySeq
+            .unsafeWrapArray(additionalArgs)
+            .map(arg =>
+              try {
+                Integer.valueOf(arg)
```

Comment:
This is surprising, non-Unix like behavior. Turns out we are OK with strings, following snippet works without `Integer` conversion:
```
polyglot java import java.lang.Integer

fac n =
    facacc n v = if n <= 1 then v else @Tail_Call facacc n-1 n*v
    
    res = facacc n 1
    res
    
main number =
    fac <| Integer.valueOf number
```
anyway this concept of arguments needs a bit of re-think. If this is what I get when providing wrong number of arguments:
```bash
$ ./runner --run engine/runner-native/src/test/resources/Factorial.enso
org.enso.interpreter.runtime.callable.function.Function@179b94be
$ ./runner --run engine/runner-native/src/test/resources/Factorial.enso 41 1
Execution finished with an error: Type error: expected a function, but got Integer.
```
probably we should pass in `Vector[String]`.